src/app/calls.ts

```diff
@@ -0,0 +1,71 @@
+import {AppCallResponse, AppSelectOption} from 'mattermost-redux/types/apps';
+
+class Calls {
+    // createForm returns a form response to create a ticket from a post
+    createForm(message: string): AppCallResponse {
+        const call: AppCallResponse = {
+            type: 'form',
+            form: {
+                title: 'Create Zendesk Ticket',
+                header: 'Create a Zendesk from Mattermost by filling in the required fields and clicking the submit button. Addition text can be added in the `Optional message` form.',
+                footer: 'Message modal form footer',
+                fields: [
+                    {
+                        name: 'subject',
+                        type: 'text',
+                        is_required: true,
+                        hint: 'subject',
+                        modal_label: 'Subject',
+                    },
+                    {
+                        name: 'type',
+                        type: 'static_select',
+                        options: getStaticSelectOptions(['problem', 'incident', 'question', 'task']),
+                        is_required: false,
+                        hint: 'type',
+                        modal_label: 'Type',
+                    },
+                    {
+                        name: 'priority',
+                        type: 'static_select',
+                        options: getStaticSelectOptions(['urgent', 'high', 'normal', 'low']),
+                        is_required: false,
+                        hint: 'priority',
+                        modal_label: 'Priority',
+                    },
+                    {
+                        name: 'additional_message',
+                        type: 'text',
+                        hint: 'Add additional message to the Zendesk ticket',
```

Comment:
I think this may be better suited to be a description, rather than a hint.

```suggestion
                        description: 'Add additional message to the Zendesk ticket',
```

Reply:
Fair enough.  I was opting for hints because then don't grow the modal vertically.  But I agree for this text it is a description, not a hint :)